src/ray/util/process.h

```diff
@@ -0,0 +1,94 @@
+#ifndef RAY_UTIL_PROCESS_H
+#define RAY_UTIL_PROCESS_H
+
+#include <functional>  // std::equal_to, std::hash, std::less
+#include <memory>      // std::shared_ptr
+#include <utility>     // std::forward
+
+#include <boost/process/args.hpp>
+#include <boost/process/child.hpp>
+
+// We only define operators required by the standard library (<, ==, hash).
+// We declare but avoid defining the rest so that they're not used by accident.
+
+namespace std {
+template <>
+struct hash<boost::process::child> : hash<boost::process::pid_t> {
+  typedef hash<boost::process::pid_t> base_type;
+  typedef boost::process::child argument_type;
+  typedef size_t result_type;
```

Comment:
For following the code style of current codebase:
1. use `using ` instead of `typedef`
2. `base_type ` -> `BaseType`, `argument_type` -> `ArgumentType`, etc.
ps: I'd like to rename `ArgumentType`  to `ChildProcess`.

Reply:
http://www.cplusplus.com/reference/functional/hash/#types